lib/vehicle_factory.rb

```diff
@@ -0,0 +1,106 @@
+class VehicleFactory
+
+    def create_vehicles(source)
+        vehicles_array = []
+        source.each do |vehicle|
+            if vehicle.keys.include?(:dol_vehicle_id)
+                vehicles_array.concat(wa_create_vehicles(source)) # .concat combining two arrays into a single array
+            elsif vehicle.keys.include?(:reg_valid_date)
+                vehicles_array.concat(ny_create_vehicles(source))
+            end
+        end
+        vehicles_array
+    end
+
+    def ny_create_vehicles(source)
+        @ny_vehicles_array = []
+        source.each do |vehicle_details|
+            vin = vehicle_details[:vin]
+            year = vehicle_details[:model_year]
+            make = vehicle_details[:make]
+            model = vehicle_details[:body_type]
+            registration_date = vehicle_details[:reg_valid_date]
+    
+            ny_vehicle = Vehicle.new({
+                vin: vin,
+                year: year,
+                make: make,
+                model: model,
+                engine: nil,
+                registration_date: registration_date,
+                plate_type: nil
+            })
+            @ny_vehicles_array << ny_vehicle
+        end
+        @ny_vehicles_array
```

Comment:
I would have an initialize method that actually declares these instance variables, so instead of having line 16 in this method, have it in the initialize. This is helpful because when we always declare our instance variables in the initialize method, we always know where to look for those instance variables.